qiskit/circuit/instruction.py

```diff
@@ -129,6 +129,45 @@ def __eq__(self, other):
 
         return True
 
+    def soft_compare(self, other: 'Instruction') -> bool:
+        """
+        Soft comparison between gates. Their names, number of qubits, and classical
+        bit numbers must match. The number of parameters must match. Each parameter
+        is compared. If one is a ParameterExpression then it is not taken into
+        account.
+
+        Args:
+            other (instruction): other instruction.
+
+        Returns:
+            bool: are self and other equal up to parameter expressions.
+        """
+        if self.name != other.name or \
+                other.num_qubits != other.num_qubits or \
+                other.num_clbits != other.num_clbits or \
+                len(self.params) != len(other.params):
+            return False
+
+        for self_param, other_param in zip_longest(self.params, other.params):
+            if isinstance(self_param, ParameterExpression) or \
+                    isinstance(other_param, ParameterExpression):
+                continue
+            elif isinstance(self_param, numpy.ndarray) and \
```

Comment:
Lint is failing because this can be
```suggestion
            if isinstance(self_param, numpy.ndarray) and \
```